azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/ManagementChannel.java

```diff
@@ -0,0 +1,170 @@
+    }
+	
+	public CompletableFuture<Map<String, Object>> doManagementRequestResponse(final ReactorDispatcher dispatcher, final Map<String, String> request)
+	{
+		final Message requestMessage = Proton.message();
+        final ApplicationProperties applicationProperties = new ApplicationProperties(request);
+        requestMessage.setApplicationProperties(applicationProperties);
+        // no body required
+        
+        CompletableFuture<Map<String, Object>> resultFuture = new CompletableFuture<Map<String, Object>>();
+        
+        this.innerChannel.runOnOpenedObject(dispatcher,
+                new IOperationResult<RequestResponseChannel, Exception>() {
+                    @Override
+                    public void onComplete(final RequestResponseChannel result) {
+                        result.request(dispatcher, requestMessage,
+                                new IOperationResult<Message, Exception>() {
+                                    @Override
+                                    public void onComplete(final Message response) {
+                                        final int statusCode = (int)response.getApplicationProperties().getValue().get(ClientConstants.PUT_TOKEN_STATUS_CODE);
+                                        final String statusDescription = (String)response.getApplicationProperties().getValue().get(ClientConstants.PUT_TOKEN_STATUS_DESCRIPTION);
+
+                                        if (statusCode == AmqpResponseCode.ACCEPTED.getValue() || statusCode == AmqpResponseCode.OK.getValue()) {
+                                            if (response.getBody() != null) {
+                                                resultFuture.complete((Map<String, Object>)((AmqpValue)response.getBody()).getValue());
+                                            }
+                                        } 
+                                        else {
+                                            this.onError(ExceptionUtil.amqpResponseCodeToException(statusCode, statusDescription));
+                                        }
+                                    }
+
+                                    @Override
+                                    public void onError(final Exception error) {
+                                    	resultFuture.completeExceptionally(error);
+                                    }
+                                });
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        resultFuture.completeExceptionally(error);
+                    }
+                });
+        
+		return resultFuture;
+	}
+
+    public void close(final ReactorDispatcher reactorDispatcher, final IOperationResult<Void, Exception> closeCallback) {
+        this.innerChannel.close(reactorDispatcher, closeCallback);
+    }
+
+    private class OpenRequestResponseChannel implements IOperation<RequestResponseChannel> {
```

Comment:
>OpenRequestResponseChannel [](start = 18, length = 26)

can you please parameterize and consolidate open and close classes to a common place - which can be consumed by both cbs and mgmt ...